server/src/main/java/com/linecorp/centraldogma/server/internal/storage/repository/DefaultMetaRepository.java

```diff
@@ -73,90 +82,232 @@ public org.eclipse.jgit.lib.Repository jGitRepository() {
-        final Entry<?> e = entries.get(PATH_CREDENTIALS);
-        if (e == null) {
-            return Collections.emptyList();
-        }
+    private List<MirrorCredential> parseCredentials(Map<String, Entry<?>> entries)
+            throws JsonProcessingException {
+        return entries.entrySet().stream()
+                      .map(entry -> {
+                          try {
+                              //noinspection unchecked
+                              return parseCredential(entry.getKey(), (Entry<JsonNode>) entry.getValue());
+                          } catch (JsonProcessingException e) {
+                              return Exceptions.throwUnsafely(e);
+                          }
+                      })
+                      .collect(toImmutableList());
+    }
 
-        final JsonNode credentialsJson = (JsonNode) e.content();
-        if (!credentialsJson.isArray()) {
-            throw new RepositoryMetadataException(
-                    PATH_CREDENTIALS + " must be an array: " + credentialsJson.getNodeType());
+    private MirrorCredential parseCredential(String credentialFile, Entry<JsonNode> entry)
+            throws JsonProcessingException {
+        final JsonNode credentialJson = entry.content();
+        if (!credentialJson.isObject()) {
+            throw newInvalidJsonTypeException(credentialFile, credentialJson);
         }
+        return Jackson.treeToValue(credentialJson, MirrorCredential.class);
+    }
 
-        if (credentialsJson.size() == 0) {
-            return Collections.emptyList();
-        }
+    private RepositoryMetadataException newInvalidJsonTypeException(String fileName, JsonNode credentialJson) {
+        return new RepositoryMetadataException(parent().name() + '/' + name() + fileName +
+                                               " must be an object: " + credentialJson.getNodeType());
+    }
 
-        final ImmutableList.Builder<MirrorCredential> builder = ImmutableList.builder();
-        for (JsonNode c : credentialsJson) {
-            final MirrorCredential credential = Jackson.treeToValue(c, MirrorCredential.class);
-            if (credential == null) {
-                throw new RepositoryMetadataException(PATH_CREDENTIALS + " contains null.");
+    private CompletableFuture<Map<String, Entry<?>>> find(String filePattern) {
+        return find(Revision.HEAD, filePattern, ImmutableMap.of());
+    }
+
+    @Override
+    public CompletableFuture<Command<CommitResult>> createPushCommand(MirrorDto mirrorDto, Author author,
+                                                                      boolean update) {
+        validateMirror(mirrorDto);
+        if (update) {
+            final String summary = "Update the mirror '" + mirrorDto.id() + '\'';
+            return mirror(mirrorDto.id()).thenApply(mirror -> {
+                // Perform the update operation only if the mirror exists.
```

Comment:
Isn't this called even when the mirror wasn't there?

Reply:
If no mirror is found, `EntryNotFoundException` will be raised. Did you mean something else?

Reply:
I missed that. 😓 Please forget about the comment.